buildkite/resource_team_member.go

```diff
@@ -3,174 +3,208 @@ package buildkite
+			resp.Diagnostics.AddError(
+				"Unable to get team member",
+				"Error getting team member: nil response",
+			)
+			return
+		}
+		updateTeamMemberResourceState(&state, *teamMemberNode)
+		resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	}
+}
 
-	return nil
+func (tm *teamMemberResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
 }
 
-// UpdateTeamMember a team members role within a Buildkite team
-func UpdateTeamMember(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	client := m.(*Client)
+func (tm *teamMemberResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var id, role string
 
-	var mutation struct {
-		TeamMemberUpdate struct {
-			TeamMember TeamMemberNode
-		} `graphql:"teamMemberUpdate(input: {id: $id, role: $role})"`
-	}
+	// Obtain team member's ID from state, new role from plan
+	resp.Diagnostics.Append(req.State.GetAttribute(ctx, path.Root("id"), &id)...)
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("role"), &role)...)
 
-	vars := map[string]interface{}{
-		"id":   d.Id(),
-		"role": TeamMemberRole(d.Get("role").(string)),
+	if resp.Diagnostics.HasError() {
+		return
 	}
 
-	err := client.graphql.Mutate(context.Background(), &mutation, vars)
+	log.Printf("Updating team member %s with role %s ...", id, role)
+	apiResponse, err := updateTeamMember(tm.client.genqlient, id, TeamMemberRole(role))
+
 	if err != nil {
-		return diag.FromErr(err)
+		resp.Diagnostics.AddError(
+			"Unable to update Team member",
+			fmt.Sprintf("Unable to update Team member: %s", err.Error()),
+		)
+		return
 	}
 
-	updateTeamMember(d, &mutation.TeamMemberUpdate.TeamMember)
-
-	return nil
+	// Update state with revised role
+	newRole := types.StringValue(string(apiResponse.TeamMemberUpdate.TeamMember.Role))
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("role"), newRole)...)
 }
 
-// DeleteTeamMember removes a user from a Buildkite team
-func DeleteTeamMember(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	client := m.(*Client)
-	var mutation struct {
-		TeamDelete struct {
-			DeletedTeamMemberId graphql.String `graphql:"deletedTeamMemberID"`
-		} `graphql:"teamMemberDelete(input: {id: $id})"`
-	}
+func (tm *teamMemberResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var id string
+
+	// Obtain team member's ID from state
+	resp.Diagnostics.Append(req.State.GetAttribute(ctx, path.Root("id"), &id)...)
 
-	vars := map[string]interface{}{
-		"id": graphql.ID(d.Id()),
+	if resp.Diagnostics.HasError() {
+		return
 	}
 
-	err := client.graphql.Mutate(context.Background(), &mutation, vars)
+	log.Printf("Deleting team member with ID %s ...", id)
+	_, err := deleteTeamMember(tm.client.genqlient, id)
+
```

Comment:
Why not just `_, err := deleteTeamMember(tm.client.genqlient, state.ID.ValueString())`?

Reply:
Adjusting in latest commit